File: fel-adapter/src/de/proskor/fel/impl/EventRepositoryImpl.scala
package de.proskor.fel.impl

import java.util.{List => JavaList}
import scala.collection.JavaConversions.seqAsJavaList
import de.proskor.automation.Element
import de.proskor.automation.Package
import de.proskor.automation.Repository
import de.proskor.fel.container.EventTypeContainer
import de.proskor.fel.EventRepository
import de.proskor.fel.view.ViewRepository
import de.proskor.fel.view.View
import de.proskor.automation.Diagram

class EventRepositoryImpl(repository: Repository) extends EventRepository with ViewRepository {
  private def fel: Package = {
    val model = repository.models.headOption getOrElse repository.models.add("Model", "Package")
    model.packages.find(_.name == "FEL") getOrElse model.packages.add("FEL", "Package")
  }

  override def getEventTypeContainers: JavaList[EventTypeContainer] = {
    repository.models.flatMap(allObjects).map(new EventTypeContainerImpl(_)).toList
  }

  override def getViews: JavaList[View] = allDiagrams(repository.models.head) map {
    case diagram: Diagram if diagram.stereotype == "CFT" => new ComponentFaultTreeImpl(diagram)
    case diagram: Diagram => new ArchitecturalViewImpl(diagram)
  }

  private def allDiagrams(pkg: Package): Seq[Diagram] =
    pkg.diagrams.toSeq ++ pkg.packages.flatMap(allDiagrams)

  private def allObjects(pkg: Package): Seq[Element] =
    pkg.elements.filter(_.stereotype == "EventTypeContainer").toSeq ++ pkg.packages.flatMap(allObjects)
}

File: ea-fel/src/de/proskor/fel/model/impl/stub/EventRepositoryImpl.java
package de.proskor.fel.model.impl.stub;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

import de.proskor.fel.model.EventRepository;
import de.proskor.fel.model.container.EventTypeContainer;
import de.proskor.fel.model.view.ArchitecturalView;
import de.proskor.fel.model.view.ComponentFaultTree;
import de.proskor.fel.model.view.View;
import de.proskor.fel.model.view.ViewRepository;

public class EventRepositoryImpl implements EventRepository, ViewRepository {
	private List<EventTypeContainer> containers = new ArrayList<EventTypeContainer>();
	private List<ArchitecturalView> archViews = new ArrayList<ArchitecturalView>();
	private List<ComponentFaultTree> cfts = new ArrayList<ComponentFaultTree>();

	@Override
	public List<EventTypeContainer> getEventTypeContainers() {
		return this.containers;
	}

	@Override
	public List<? extends View> getViews() {
		List<View> result = new LinkedList<View>(this.archViews);
		result.addAll(this.cfts);
		return result;
	}
}
